### ISSUE TITLE
Fix the unsound parts of Object.values/Object.entries

### ISSUE THREAD

**Issue by guillaumebrunerie** [total-typescript/ts-reset#60]
It's a great idea not to "fix" `Object.keys` as it is not broken, I fully support that! But you could push the reasoning further and actually fix the various broken parts of `Object.values` and `Object.entries`.

For instance it can introduce unexpected `any`s:
```typescript
// Return type: any[] | null
const f = (x: unknown) => {
    if (x) {
        return Object.values(x);
    }
    return null;
}

// Return type: [string, any][] | null
const g = (x: unknown) => {
    if (x) {
        return Object.entries(x);
    }
    return null;
}
```

And it also does not respect structural typing and the possibility of having extra properties:
```typescript
// Return type: number[], but x could have other properties with different types, should really be unknown[]
const h = (x: {a: number}) => Object.values(x);

// Return type: [string, number][] which is wrong in general for the same reason.
const i = (x: {a: number}) => Object.entries(x);
```

**Comment by daniellwdb** [comment]
https://github.com/niieani/nesity/tree/main/packages/types#fromentries

**Comment by guillaumebrunerie** [comment]
Indeed, `Object.fromEntries` can also create unexpected anys (if that's what you mean), and should be fixed as well. The special case of constant readonly tuples seems extremely narrow, though, and not really worth it in my opinion.

**Comment by muuvmuuv** [comment]
Oh yes, that would be awesome I generally use this here:

```ts
declare interface ObjectConstructor {
	keys<T extends object>(o: T): Extract<keyof T, string>[]
	values<T extends object>(o: T): T[Extract<keyof T, string>][]
	entries<T extends object>(o: T): [Extract<keyof T, string>, T[Extract<keyof T, string>]][]
}
```

(partly copied from type-fest `Entries<T>`)

**Comment by guillaumebrunerie** [comment]
> Oh yes, that would be awesome I generally use this here:
> 
> declare interface ObjectConstructor {
> 	keys<T extends object>(o: T): Extract<keyof T, string>[]
> 	values<T extends object>(o: T): T[Extract<keyof T, string>][]
> 	entries<T extends object>(o: T): [Extract<keyof T, string>, T[Extract<keyof T, string>]][]
> }
> (partly copied from type-fest `Entries<T>`)

What I am suggesting to do is the exact opposite. What you're suggesting to use is just completely unsound.
